src/prpy/base/robot.py

```diff
@@ -284,15 +284,33 @@ def ViolatesVelocityLimits(self, traj):
         return False
 
     def _PlanWrapper(self, planning_method, args, kw_args):
-        # Call the planner.
-        traj = planning_method(self, *args, **kw_args)
 
-        # Strip inactive DOFs from the trajectory.
+        # Call the planner.
         config_spec = self.GetActiveConfigurationSpecification()
-        openravepy.planningutils.ConvertTrajectorySpecification(traj, config_spec)
+        result = planning_method(self, *args, **kw_args)
+
+        # Define the post processing steps for the trajectory.
+        def postprocess_trajectory(traj, kw_args):
+
+            # Strip inactive DOFs from the trajectory.
+            openravepy.planningutils.ConvertTrajectorySpecification(
+                traj, config_spec
+            )
+
+            # Optionally execute the trajectory.
+            if 'execute' not in kw_args or kw_args['execute']:
+                return self.ExecuteTrajectory(traj, **kw_args)
+            else:
+                return traj
```

Comment:
Currently, `ExecuteTrajectory` can raise `TrajectoryExecutionError` exceptions if a `timeout` is specified. This is an unfortunate side-effect of the current blocking implementation.

I'd prefer to make `ExecuteTrajectory` always use a future internally. The effect of passing `defer=False`, if a timeout is specified, is simply to wait on the future before returning. The downside of this approach is that PrPy must depend on Trollius, even if `defer` is never being passed.